beacon-chain/powchain/service.go

```diff
@@ -119,6 +121,12 @@ var (
 // NewWeb3Service sets up a new instance with an ethclient when
 // given a web3 endpoint as a string in the config.
 func NewWeb3Service(ctx context.Context, config *Web3ServiceConfig) (*Web3Service, error) {
+	return NewWeb3ServiceWithDbSupport(ctx, config, false)
+}
+
+// NewWeb3ServiceWithDbSupport sets up a new instance with an ethclient when
+// given a web3 endpoint as a string in the config.
+func NewWeb3ServiceWithDbSupport(ctx context.Context, config *Web3ServiceConfig, initFromDb bool) (*Web3Service, error) {
```

Comment:
Why do we need another method? Shouldn't the default be with database support?

Reply:
Should I Keep one function with the same signature and implement db sync with no boolean?

Reply:
I kept the original function signature and implement read from db in it